grails-app/controllers/au/org/ala/biocollect/BioActivityController.groovy

```diff
@@ -1801,6 +1804,171 @@ class BioActivityController {
+                            schema = @Schema(
+                                    type = "string",
+                                    defaultValue = "desc"
+                            )
+                    ),
+                    @Parameter(
+                            name = "status",
+                            in = ParameterIn.QUERY,
+                            description = "Project status",
+                            schema = @Schema(
+                                    type = "string",
+                                    defaultValue = "active"
+                            )
+                    )
+            ],
+            responses = [
+                    @ApiResponse(
+                            responseCode = "200",
+                            content = @Content(
+                                    mediaType = "application/json",
+                                    schema = @Schema(
+                                            implementation = Map.class
+                                    )
+                            ),
+                            headers = [
+                                    @Header(name = 'Access-Control-Allow-Headers', description = "CORS header", schema = @Schema(type = "String")),
+                                    @Header(name = 'Access-Control-Allow-Methods', description = "CORS header", schema = @Schema(type = "String")),
+                                    @Header(name = 'Access-Control-Allow-Origin', description = "CORS header", schema = @Schema(type = "String"))
+                            ]
+                    ),
+                    @ApiResponse(responseCode = "400", description = "Bad request"),
+                    @ApiResponse(responseCode = "401", description = "Unauthorized"),
+                    @ApiResponse(responseCode = "500", description = "Internal server error")
+            ],
+            security = @SecurityRequirement(name = "auth")
+    )
+    @Path("ws/bioactivity/data/records")
+    def listRecordsForDataResourceId(){
+        String userId = userService.getCurrentUserId()
+        GrailsParameterMap queryParams = constructDefaultSearchParams(params)
+
+        if (!userId) {
+            render text: [message: "Access denied: User not authorised."] as JSON, status: HttpStatus.SC_UNAUTHORIZED, contentType: ContentType.APPLICATION_JSON
+        } else if (!queryParams.projectId) {
+            render text: [message: "No project associated with the activity."] as JSON, status: HttpStatus.SC_BAD_REQUEST, contentType: ContentType.APPLICATION_JSON
+        } else if (projectService.canUserEditProject(userId, queryParams.projectId, false)) {
+            Map project = projectService.get(queryParams.projectId)
+            if (!project.dataResourceId) {
+                render text: [message: "Data resource Id is required."] as JSON, status: HttpStatus.SC_BAD_REQUEST, contentType: ContentType.APPLICATION_JSON
```

Comment:
"Only data of ALA harvested projects can be accessed."

Reply:
@temi so is this a wrong validation?
} else if (projectService.canUserEditProject(userId, queryParams.projectId, false)) {

sorry unsure what is a ALA harvested projects.

Reply:
or do you want me to add this check?
if(project?.alaHarvest) {}